packages/react/test/reduxStateAttachments.test.ts

```diff
@@ -0,0 +1,105 @@
+import * as Sentry from '@sentry/browser';
+import * as Redux from 'redux';
+
+import { createReduxEnhancer } from '../src/redux';
+
+const mockAddBreadcrumb = jest.fn();
+const mockSetContext = jest.fn();
+
+jest.mock('@sentry/browser', () => ({
+  ...jest.requireActual('@sentry/browser'),
+}));
+
+afterEach(() => {
+  mockAddBreadcrumb.mockReset();
+  mockSetContext.mockReset();
+});
+
+describe('Redux State Attachments', () => {
+  it('attaches Redux state to Sentry scope', () => {
+    const enhancer = createReduxEnhancer();
+
+    const initialState = {
+      value: 'initial',
+    };
+
+    const store = Redux.createStore((state = initialState) => state, enhancer);
+
+    const updateAction = { type: 'UPDATE_VALUE', value: 'updated' };
+
+    store.dispatch(updateAction);
+
+    const error = new Error('test');
+    Sentry.captureException(error);
+
+    Sentry.configureScope(scope => {
+      expect(scope.getAttachments()).toContainEqual(
+        expect.objectContaining({
+          filename: 'redux_state.json',
+          data: JSON.stringify({
+            value: 'updated',
+          }),
+        }),
+      );
+    });
+  });
```

Comment:
We can access the attachments on the scope via `Sentry.getCurrentHub().getScope().getAttachments()` - let's do that for the rest of the uses of `configureScope` as well.

Reply:
i tried that earlier but i wasn't running as expected  

<img width="779" alt="Screenshot 2023-09-07 at 6 28 51 PM" src="https://github.com/getsentry/sentry-javascript/assets/101856674/0e5d4133-ee47-4c62-924c-bca78ef0b3e9">

I also tried a variation of it thinking it might work but don't get it why is it not working cause under the hood both the things are same 

<img width="744" alt="Screenshot 2023-09-07 at 6 41 10 PM" src="https://github.com/getsentry/sentry-javascript/assets/101856674/0327aced-d4d8-4906-8846-5ec1898d6c79">

Reply:
I discovered today that I made an error when writing the tests. They currently pass for every scenario. I'll investigate this issue further and let you know how I'm doing.

Reply:
> I also tried a variation of it thinking it might work but don't get it why is it not working cause under the hood both the things are same

Regarding this 🫣 situation, it seems both versions had the same behavior: one passed while the other failed. The root issue was in the callback within the code snippet provided. The callback was pushed onto a stack but never executed 🫠, causing the test case to never fail.

```typescript
 Sentry.configureScope(scope => {
      expect(scope.getAttachments()).toContainEqual(
        expect.objectContaining({
          filename: 'redux_state.json',
          data: JSON.stringify({
            value: 'updated',
          }),
        }),
      );
    });
```